src/emmental/task.py

```diff
@@ -19,14 +19,35 @@
 class EmmentalTaskFlowAction:
     """An action to execute in a EmmentalTask task_flow.
 
+    EmmentalTaskFlowAction is the object that populate the task_flow sequence.
+    It has three attributes: name, module_name and inputs where name is the name of the
+    action, module_name is the module name used in this action and inputs is the inputs
+    to the action. By introducing a class for specifying actions in the task_flow,
+    we standardize its definition. Moreover, EmmentalTaskFlowAction enables more user
+    flexibility in specifying a task flow as we can now support a wider-range of
+    formats for the input attribute of a task_flow as follow:
+
+    1. It now supports str as inputs (e.g., inputs="input1") which means take the
+    input1's output as input for current action.
+
+    2. It also support None as inputs which will take all modules' output as input.
```

Comment:
Where is the handling on index being None in get_data_from_output_dict?

Reply:
But the comments help a lot here.

Reply:
It's been handled here: https://github.com/senwu/emmental/blob/master/src/emmental/model.py#L300